[docs] add diagrams and fix broken links (#16343)
MystenLabs/sui: docs/content/guides/developer/sui-101/shared-owned.mdx

### Owned objects

[Code Sample](https://github.com/MystenLabs/sui/blob/93e6b4845a481300ed4a56ab4ac61c5ccb6aa008/examples/move/escrow/sources/owned.move)

The protocol for swapping via escrow implemented using owned objects starts with both parties locking their respective objects. 

    ![Escrow Step 1](../images/escrow-step1.png)

> This will be used to prove that the object has not been tampered with after the swap has been agreed. If either party does not want to proceed at this stage, they just unlock their object.

Reviewer: ```suggestion
This is used to prove that the object has not been tampered with after the swap has been agreed to. If either party doesn't want to proceed at this stage, they just unlock their object.
```

Assuming both parties are happy to continue, the next step requires both parties to swap the keys.

    ![Escrow Step 2](../images/escrow-step2.png)

> We then introduce a third party to act as custodian. The custodian holds objects that are waiting for their counterparts to arrive and when they arrive, it matches them up to complete the swap.

Reviewer: ```suggestion
A third party acts as custodian. The custodian holds objects that are waiting for their counterparts to arrive and when they arrive, it matches them up to complete the swap.
```
